starknet-rpc-test/get_block_hash_and_number.rs

```diff
@@ -1,19 +1,52 @@
 extern crate starknet_rpc_test;
 
+use starknet_accounts::SingleOwnerAccount;
+use starknet_core::chain_id;
 use starknet_ff::FieldElement;
 use starknet_providers::Provider;
-use starknet_rpc_test::{ExecutionStrategy, MadaraClient};
+use starknet_rpc_test::constants::{ARGENT_CONTRACT_ADDRESS, MINT_AMOUNT, SIGNER_PRIVATE};
+use starknet_rpc_test::utils::transfer_tokens;
+use starknet_rpc_test::{BlockCreation, ExecutionStrategy, MadaraClient};
+use starknet_signers::{LocalWallet, SigningKey};
 
 #[tokio::test]
-async fn work_ok_at_initialization() -> Result<(), anyhow::Error> {
+async fn work_ok_at_start_and_with_new_blocks() -> Result<(), anyhow::Error> {
     let madara = MadaraClient::new(ExecutionStrategy::Native).await;
     let rpc = madara.get_starknet_client();
 
     assert_eq!(
         rpc.block_hash_and_number().await?.block_hash,
-        FieldElement::from_hex_be("0x031ebd02657f940683ae7bddf19716932c56d463fc16662d14031f8635df52ad").unwrap()
+        FieldElement::from_hex_be("0x035d394a3808b432702261c2d66f1cf2f111dd0c691ad2a8246b86a5088f73fa").unwrap()
     );
     assert_eq!(rpc.block_hash_and_number().await?.block_number, 0);
 
+    madara.create_block(vec![], BlockCreation::default()).await?;
+    assert_eq!(
+        rpc.block_hash_and_number().await?.block_hash,
+        FieldElement::from_hex_be("0x001d68e058e03162e4864ef575445c38deea4fad6b56974ef9012e8429c2e7b9").unwrap()
+    );
+    assert_eq!(rpc.block_hash_and_number().await?.block_number, 1);
+
+    let signer = LocalWallet::from(SigningKey::from_secret_scalar(FieldElement::from_hex_be(SIGNER_PRIVATE).unwrap()));
+    let argent_account_address = FieldElement::from_hex_be(ARGENT_CONTRACT_ADDRESS).expect("Invalid Contract Address");
+    let account = SingleOwnerAccount::new(rpc, signer, argent_account_address, chain_id::TESTNET);
+
+    madara
+        .create_block(
+            vec![transfer_tokens(
+                &account,
+                argent_account_address,
+                FieldElement::from_hex_be(MINT_AMOUNT).expect("Invalid Mint Amount"),
+                None,
+            )],
+            BlockCreation::new(None, true),
+        )
+        .await?;
```

Comment:
Why are you sending a tx? What behaviour are you testing with this regarding `block_hash_and_number`.

Reply:
> Return the correct block number at blocks and hash after diverse blocks with diverse tx are created